core/lib/zksync_core/src/sync_layer/snapshots.rs

```diff
@@ -0,0 +1,496 @@
+use crate::metadata_calculator::{MetadataCalculator, MetadataCalculatorConfig};
+use crate::state_keeper::{L1BatchExecutorBuilder, MainBatchExecutorBuilder, ZkSyncStateKeeper};
+use crate::sync_layer::fetcher::MainNodeFetcherCursor;
+use crate::sync_layer::snapshots::SnapshotApplierError::*;
+use crate::sync_layer::{ActionQueue, ExternalIO, MainNodeClient, SyncState};
+use anyhow::Context;
+use multivm::vm_1_3_2::zk_evm_1_3_3::ethereum_types::Address;
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::path::Path;
+use tokio::sync::watch;
+use zksync_dal::connection::DbVariant;
+use zksync_dal::{ConnectionPool, StorageProcessor};
+use zksync_merkle_tree::recovery::{MerkleTreeRecovery, RecoveryEntry};
+use zksync_merkle_tree::RocksDBWrapper;
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::block::{BlockGasCount, MiniblockHeader};
+use zksync_types::snapshots::{
+    AppliedSnapshotStatus, SnapshotFactoryDependencies, SnapshotHeader, SnapshotStorageLog,
+    SnapshotStorageLogsChunk, SnapshotStorageLogsStorageKey,
+};
+use zksync_types::{
+    L1BatchNumber, L2ChainId, ProtocolVersionId, StorageKey, StorageLog, StorageLogKind, H256,
+};
+
+pub struct StateKeeperConfig {
+    pub connection_pool: ConnectionPool,
+    pub state_keeper_db_path: String,
+    pub l2_erc20_bridge_addr: Address,
+    pub chain_id: L2ChainId,
+    pub main_node_url: String,
+    pub enum_index_migration_chunk_size: usize,
+}
+pub struct SnapshotApplier<'a, 'b, 'd> {
+    connection_pool: &'a ConnectionPool,
+    client: &'a dyn MainNodeClient,
+    recovery: MerkleTreeRecovery<'b, RocksDBWrapper>,
```

Comment:
Lifetime here can be replaced in favor of `'static`. I probably need to get rid of this lifetime anyway.